src/util/comparison.ts

```diff
@@ -46,6 +46,35 @@ export function selectMostSimilarChapter(original: Chapter, options: Chapter[]):
   return null;
 }
 
+function consolidateAndSortChapters(chapterList: Chapter[]): Chapter[] {
+  const grouped: { [index: string]: Chapter[] } = {};
+  chapterList.forEach((chapter: Chapter) => {
+    const key = chapter.chapterNumber === '' ? chapter.sourceId : chapter.chapterNumber;
+
+    if (grouped[key] === undefined) {
+      grouped[key] = [];
+    }
+
+    grouped[key].push(chapter);
+  });
+
+  const chapters: Chapter[] = [];
+  Object.keys(grouped).forEach((key) => {
+    const groupedChapters = grouped[key];
+
+    let chapter = groupedChapters.find((chap) => chap.read);
+    if (chapter === undefined) {
+      [chapter] = groupedChapters;
+    }
```

Comment:
The `[chapter]` assignment here looks pretty weird, it seems to work but I don't really understand why. I think this section is equivalent to just:
```
let chapter = groupedChapters.find(chap => chap.read) || groupedChapters[0];
```

Reply:
The reason I did this was because the linter kept complaining at me, so I assumed it was because of the project level linter config 😅 

In hindsight it was probably just my VSCode extension 🤷🏾